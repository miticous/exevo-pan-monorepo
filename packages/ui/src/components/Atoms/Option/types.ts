import { HTMLAttributes } from 'react'

export interface OptionProps {
  children: string
  value?: string
  highlighted?: boolean
  onClick?: (option: Option) => void
  onMouseDown?: (event: React.MouseEvent) => void
  props?: HTMLAttributes<HTMLOptionElement>
}

export interface OptionStyleProps {
  highlighted?: boolean
}
