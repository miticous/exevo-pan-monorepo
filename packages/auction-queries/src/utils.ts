export const deserializeFilterOptions = ({
  vocation,
  pvp,
  battleye,
  location,
  serverSet,
  skillKey,
  imbuementsSet,
  charmsSet,
  itemSet,
  questSet,
  outfitSet,
  storeOutfitSet,
  mountSet,
  storeMountSet,
  achievementSet,
  ...primitiveOptions
}: SerializedFilterOptions): FilterOptions => ({
  ...primitiveOptions,
  vocation: new Set<VocationOptions>(vocation as VocationOptions[]),
  pvp: new Set<PvpOptions>(pvp as PvpOptions[]),
  battleye: new Set<boolean>(battleye),
  location: new Set<LocationOptions>(location as LocationOptions[]),
  serverSet: new Set<string>(serverSet),
  skillKey: new Set<SkillOptions>(skillKey as SkillOptions[]),
  imbuementsSet: new Set<string>(imbuementsSet),
  charmsSet: new Set<string>(charmsSet),
  itemSet: new Set<string>(itemSet),
  questSet: new Set<string>(questSet),
  outfitSet: new Set<string>(outfitSet),
  storeOutfitSet: new Set<string>(storeOutfitSet),
  mountSet: new Set<string>(mountSet),
  storeMountSet: new Set<string>(storeMountSet),
  achievementSet: new Set<string>(achievementSet),
})

export const DEFAULT_FILTER_OPTIONS: SerializedFilterOptions = {
  nicknameFilter: '',
  vocation: [],
  pvp: [],
  battleye: [],
  location: [],
  serverSet: [],
  minLevel: 8,
  maxLevel: 2000,
  minSkill: 10,
  maxSkill: 150,
  sex: false,
  skillKey: [],
  imbuementsSet: [],
  charmsSet: [],
  itemSet: [],
  rareNick: false,
  questSet: [],
  addon: 3,
  outfitSet: [],
  storeOutfitSet: [],
  mountSet: [],
  storeMountSet: [],
  achievementSet: [],
  soulwarAvailable: false,
}
