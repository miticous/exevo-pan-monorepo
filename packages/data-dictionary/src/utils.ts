export const lowerCaseKeys = (object: ScrapingTokens): ScrapingTokens => {
  const newObject: ScrapingTokens = {}
  Object.keys(object).forEach((key) => {
    newObject[key.toLowerCase()] = object[key]
  })

  return newObject
}

export const nameableToScrapingTokens = (
  outfitArray: { name: string }[],
): ScrapingTokens => {
  const names = outfitArray.map(({ name }) => name)

  const scrapingTokens: ScrapingTokens = {}
  names.forEach((name) => {
    scrapingTokens[name] = name
  })

  return scrapingTokens
}

export const dictionaryFactory = (keyArray: string[]): Dictionary => {
  const dictionaryObject: Dictionary = {}

  keyArray.forEach((value, index) => {
    dictionaryObject[index] = value
    dictionaryObject[value] = index
  })

  return dictionaryObject
}
