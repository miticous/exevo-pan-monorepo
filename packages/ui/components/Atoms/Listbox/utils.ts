export const indexToId = (
  index: number | undefined,
  suffix?: string,
): string | undefined =>
  index === undefined
    ? undefined
    : `listbox-item-${index}${suffix ? `-${suffix}` : ''}`
