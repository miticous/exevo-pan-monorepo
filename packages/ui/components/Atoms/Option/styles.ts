import styled, { css } from 'styled-components'
import { OptionStyleProps } from './types'

const highlightStyle = css`
  background-color: var(--primaryVariant);
`

export const Option = styled.option<OptionStyleProps>`
  padding: 8px 12px;
  width: 100%;
  background-color: var(--surface);
  transition: background-color 0.2s ease-out;

  font-size: 12px;
  font-weight: 300;
  color: var(--onSurface);

  cursor: pointer;

  ${({ highlighted }) => highlighted && highlightStyle}

  &:hover {
    ${highlightStyle}
  }
`
