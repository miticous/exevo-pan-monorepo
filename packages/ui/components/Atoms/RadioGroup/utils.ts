export const indexToId = (index: number | undefined): string | undefined =>
  index === undefined ? undefined : `radio-item-${index}`
