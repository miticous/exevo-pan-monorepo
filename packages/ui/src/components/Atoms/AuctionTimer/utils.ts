import { CountdownObject } from './types'

export const MILLISECONDS_IN_A_MINUTE = 60000
export const MILLISECONDS_IN_AN_HOUR = 3600000
export const MILLISECONDS_IN_A_DAY = 86400000

export const calcCountdown = (
  startTime: number,
  endTime: number,
): CountdownObject => {
  const timeDiff = endTime - startTime

  const days = Math.floor(timeDiff / MILLISECONDS_IN_A_DAY)
  const hours = Math.floor(
    (timeDiff % MILLISECONDS_IN_A_DAY) / MILLISECONDS_IN_AN_HOUR,
  )
  const minutes = Math.floor(
    (timeDiff % MILLISECONDS_IN_AN_HOUR) / MILLISECONDS_IN_A_MINUTE,
  )
  const seconds = Math.floor((timeDiff % MILLISECONDS_IN_A_MINUTE) / 1000)

  return { timeDiff, days, hours, minutes, seconds }
}
