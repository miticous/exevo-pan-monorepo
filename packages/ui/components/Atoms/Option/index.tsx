import { memo } from 'react'
import * as S from './styles'
import { OptionProps } from './types'

const Option = ({
  children,
  value = children,
  highlighted = false,
  onClick,
  ...props
}: OptionProps): JSX.Element => (
  <S.Option
    role="option"
    value={value}
    highlighted={highlighted}
    onClick={() => onClick?.({ name: children, value })}
    {...props}
  >
    {children}
  </S.Option>
)

export default memo(Option)
