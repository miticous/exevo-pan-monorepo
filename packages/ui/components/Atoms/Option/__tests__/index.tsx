import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithProviders } from 'utils/test'
import Option from '..'

const mockedOnClick = jest.fn()

describe('<Option />', () => {
  beforeEach(() => {
    mockedOnClick.mockReset()
  })

  test('should render with correct name and value', () => {
    renderWithProviders(<Option value="actualValue">children</Option>)

    const optionElement = screen.getByRole('option')
    expect(optionElement).toHaveValue('actualValue')
    expect(screen.getByText('children')).toBeInTheDocument()
    /* expect(optionElement).toHaveStyle('background-color: #FFFFFF') */
    expect(optionElement).toMatchSnapshot()
  })

  test('should render with fallback value', () => {
    renderWithProviders(<Option>children</Option>)

    const optionElement = screen.getByRole('option')
    expect(optionElement).toHaveValue('children')
    expect(screen.getByText('children')).toBeInTheDocument()
    /* expect(optionElement).toHaveStyle('background-color: #FFFFFF') */
    expect(optionElement).toMatchSnapshot()
  })

  test('should render with highlighted style', () => {
    renderWithProviders(<Option highlighted>children</Option>)

    /* expect(screen.getByRole('option')).toHaveStyle('background-color: #C5CAE9') */
    expect(screen.getByRole('option')).toMatchSnapshot()
  })

  test('should call onClick function with an Option object', () => {
    renderWithProviders(
      <Option value="actualValue" onClick={mockedOnClick}>
        children
      </Option>,
    )

    userEvent.click(screen.getByRole('option'))
    expect(mockedOnClick).toBeCalledTimes(1)
    expect(mockedOnClick).toBeCalledWith({
      name: 'children',
      value: 'actualValue',
    })
  })

  test('should call onClick function with an Option object AND fallback value', () => {
    renderWithProviders(<Option onClick={mockedOnClick}>children</Option>)

    userEvent.click(screen.getByRole('option'))
    expect(mockedOnClick).toBeCalledTimes(1)
    expect(mockedOnClick).toBeCalledWith({
      name: 'children',
      value: 'children',
    })
  })
})
