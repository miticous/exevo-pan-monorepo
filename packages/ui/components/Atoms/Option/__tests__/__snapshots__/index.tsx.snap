// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option /> should render with correct name and value 1`] = `
<option
  class="styles__Option-sc-am72ib-0 fZyPDg"
  role="option"
  value="actualValue"
>
  children
</option>
`;

exports[`<Option /> should render with fallback value 1`] = `
<option
  class="styles__Option-sc-am72ib-0 fZyPDg"
  role="option"
  value="children"
>
  children
</option>
`;

exports[`<Option /> should render with highlighted style 1`] = `
<option
  class="styles__Option-sc-am72ib-0 eNIDEs"
  role="option"
  value="children"
>
  children
</option>
`;
